pyiron/atomistics/job/atomistic.py

```diff
@@ -629,34 +629,28 @@ def get_structure(self, iteration_step=-1, wrap_atoms=True):
         Returns:
             pyiron.atomistics.structure.atoms.Atoms: The required structure
         """
-        if not (self.structure is not None):
-            raise AssertionError()
+        if self.structure is None:
+            raise AssertionError('Structure not set')
         snapshot = self.structure.copy()
-        conditions = list()
-        if isinstance(self.output.cells, (list, np.ndarray)):
-            if len(self.output.cells) == 0:
-                conditions.append(True)
-            else:
-                conditions.append(self.output.cells[0] is None)
-        if self.output.positions is not None and self.output.cells is None:
-            conditions.append(self.output.cells is None)
-        if any(conditions):
-            snapshot.cell = None
-        elif self.output.cells is not None:
+        try:
             snapshot.cell = self.output.cells[iteration_step]
-        if self.output.positions is not None:
-            snapshot.positions = self.output.positions[iteration_step]
-        indices = self.output.indices
-        if indices is not None and len(indices) > max([iteration_step, 0]):
-            snapshot.indices = indices[iteration_step]
-        if wrap_atoms:
-            return snapshot.center_coordinates_in_unit_cell()
-        else:
+        except IndexError:
+            pass
+        try:
+            snapshot.indices = self.output.indices[iteration_step]
+        except IndexError:
+            pass
+        try:
+            if wrap_atoms:
+                snapshot.positions = self.output.positions[iteration_step]
+                return snapshot.center_coordinates_in_unit_cell()
```

Comment:
I have a mild stylistic preference to omit the `return` here so this clause is in line with the other two. `center_coordinates_in_unit_cell` looks like it will work in-place so I don't think we need a `snapshot = ...` but just to remove `return`.

Reply:
Done!